metafix/src/test/java/org/metafacture/metafix/RecordTest.java

```diff
@@ -343,20 +341,20 @@ public void shouldPreserveOrderWhenTransformingArraySubfields() {
     }
 
     @Test
-    @MetafixToDo("See https://github.com/metafacture/metafacture-fix/pull/170")
     public void shouldPreserveOrderWhenTransformingOptionalArraySubfield() {
         final Record record = new Record();
+        final String path = String.join(".", FIELD, "%s", OTHER_FIELD);
         record.put(FIELD, Value.newArray(a -> {
             a.add(Value.newHash(h -> {
                 h.put(FIELD, VALUE);
             }));
             a.add(Value.newHash(h -> {
                 h.put(FIELD, VALUE);
-                h.put(OTHER_FIELD, OTHER_VALUE);
+                // For proper transformation, we need to explicitly set the path (as in Metafix.java):
+                h.put(OTHER_FIELD, new Value(OTHER_VALUE.asString(), String.format(path, a.size() + 1)));
```

Comment:
Now I have to admit that I'm a little concerned by this. Shouldn't this be done by `Value.Array.add()` then? I don't really think it's reasonable to expect users of the API to set the paths appropriately. Especially if it's only rarely actually required.

Reply:
I agree, and as mentioned above, I currently think the best place would be in Record#add, since we have the full path information available there. If we want to set it in Array, I think we'd have to track/append the path (as I tried before). But it seems this is no regression as the test doesn't pass in master (unless I made some mistake while adapting it for master), so I'd suggest we call this a month and shift remaining problems to smaller issues / PRs.